lib/pure/math.nim

```diff
@@ -202,28 +202,39 @@ when not defined(JS):
     ## computes x to power raised of y.
     
   # C procs:
-  proc srand(seed: cint) {.importc: "srand", header: "<stdlib.h>".}
-  proc rand(): cint {.importc: "rand", header: "<stdlib.h>".}
+  when defined(windows):
+    # The "secure" random, available from Windows XP
+    # https://msdn.microsoft.com/en-us/library/sxtz2fa8.aspx
+    proc rand_s(val: var cuint) {.importc: "rand_s", header: "<stdlib.h>".}
```

Comment:
~~It does. I'm actually developing with Mingw-w64. See https://github.com/mingw-w64/mingw-w64/blame/master/mingw-w64-headers/crt/stdlib.h#L390 Seems like it has been there since 2010 at least.~~
It turns out, aggravatingly, that this is actually in mingw-w64, the 64-bit-supporting variant. It seems like the normal Mingw doesn't support it very well, at least not until very recently (see http://sourceforge.net/p/mingw/bugs/2122/; it's from late 2013). It's unfortunate, since `rand_s` is just a simple call.

I could rewrite this to work under the Microsoft compilers if you'd like, or you can close this PR. Checking for Mingw-w64 or something similar seems way too messy for the standard library.